promiseFns/filter.js

```diff
@@ -1,6 +1,9 @@
 const util = require("./util");
 module.exports = (Bluebird) => {
-    Bluebird.filter = (x, predicate, opts) => Bluebird.resolve(x).filter(predicate, opts);
```

Comment:
They are not, `filter`, `each` and `map` didn't return bluebird promise, I have tested it. 
I can move bluebird resolve wrapper to the prototype functions if it seems cleaner to you.